test/framework/toy_build.py

```diff
@@ -2707,6 +2708,25 @@ def grab_gcc_rpath_wrapper_args():
         write_file(toy_ec, toy_ec_txt)
         self.test_toy_build(ec_file=toy_ec, extra_args=['--rpath', '--experimental'], raise_error=True)
 
+    def test_toy_rpath_filter(self):
+        """Test toy build using --rpath --rpath-filter and --rpath --rpath-filter --filter-rpath-sanity-libs."""
+
+        # test sanity error when --rpath-filter is used to filter a required library
+        # In this test, libtoy.so will be linked, but not RPATH-ed due to the --rpath-filter
+        # Thus, the RPATH sanity check is expected to fail with libtoy.so not being found
+        test_ecs = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'easyconfigs', 'test_ecs')
+        toy_ec = os.path.join(test_ecs, 't', 'toy2', 'toy2-0.0.eb')
+        error_pattern = r"Sanity check failed\: Library libtoy\.so not found"
+        self.assertErrorRegex(EasyBuildError, error_pattern, self.test_toy_build, ec_file=toy_ec,
+                              extra_args=['--rpath', '--experimental', '--rpath-filter=.*libtoy.*'],
+                              name='toy2', raise_error=True, verbose=False)
+
+        # test use of --filter-rpath-sanity-libs option. In this test, we use --rpath-filter to make sure libtoy.so is
+        # not rpath-ed. Then, we use --filter-rpath-sanity-libs to make sure the RPATH sanity checks ignores
+        # the fact that libtoy.so is not found. Thus, this build should complete succesfully
+        args = ['--rpath', '--experimental', '--rpath-filter=.*libtoy.*', '--filter-rpath-sanity-libs=libtoy.so']
+        self.test_toy_build(ec_file=toy_ec, name='toy2', extra_args=args, raise_error=True)
```

Comment:
Maybe here we should also do a check on the output of `readelf -d` on the binary, and check the RPATH section with a regex (to make sure it doesn't include `libtoy`)?
That mainly makes sense if my suggestion above to also run once without `--rpath-filter` is applied.

Reply:
Hm, good idea, but I need some help with this. How do I retrieve the installdir in a test? And what is the appropriate way to run `readelf -d` in a test, do we use the `run_cmd` function for that?  (if so, I need to find where that is defined again... :))